model/versioning/Versioned.php

```diff
@@ -2126,8 +2118,11 @@ public static function prepopulate_versionnumber_cache($class, $stage, $idList =
 			$parameters = $idList;
 		}
 
-		$baseClass = ClassInfo::baseDataClass($class);
-		$stageTable = ($stage == static::DRAFT) ? $baseClass : "{$baseClass}_{$stage}";
+		$baseClass = DataObject::getSchema()->baseDataClass($class);
+		$stageTable = DataObject::getSchema()->tableName($baseClass);
+		if($stage === static::LIVE) {
```

Comment:
This is a change in logic; it will break if there are any stage other than draft or live (which, admittedly, we never do).

`if($stage !== static::DRAFT)` would preserve the original logic.